sdk/identity/identity/src/util/subscriptionUtils.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { CredentialLogger, formatError } from "./logging";
+
+/**
+ * @internal
+ */
+export function checkSubscription(logger: CredentialLogger, subscription: string): void {
+  if (!subscription.match(/^[0-9a-zA-Z-. ]+$/)) {
```

Comment:
should we leave subscription validation to the cli? 

```
az account get-access-token -s foobar
Subscription 'foobar' not found. Check the spelling and casing and try again.
```

Reply:
This validation is checking to make sure the character is correct, similar to the logic to validate "tenantID", so it should still be done in the SDK with a separate error message. The CLI validation is for non-existent subscription and it's different from our validation here.

Reply:
If it helps us make a decision, the library already validates tenant ID for this credential:

https://github.com/Azure/azure-sdk-for-js/blob/2a7059e425640a041b92e37520838c8c815464f9/sdk/identity/identity/src/util/tenantIdUtils.ts#L12

Why treat the subscription any different? By the way, I see Go uses this validation logic:

https://github.com/Azure/azure-sdk-for-go/blob/6055ff776e91e56dfac6c3629801a3dfe1fb02e9/sdk/azidentity/azure_cli_credential.go#L72

Reply:
Validation is for preventing shell injection, not ensuring correctness